### PR TITLE
LambdaAnalyzer: enable for overloaded methods with delegate args

### PR DESCRIPTION
**Pull request by DedSec256**

### CODE REVIEW

**Comment by DedSec256** [comment]
it is still necessary to support the case

```F#
type A() =
    member x.M( _: int, a: Action<int>) = ()

[<Extension>]
type Extensions() =
    [<Extension>]
    static member M(_: A, _: int, _: int) = ()

A().M(0, fun x -> ignore x) 
 
```

File: ReSharper.FSharp/src/FSharp.Psi.Features/src/Daemon/Analyzers/LambdaAnalyzer.fs [hunk 1]
```diff
@@ -96,11 +96,22 @@ type LambdaAnalyzer() =
...
+            if args.Count <= lambdaPos then false else
+            let argDecl = args.[lambdaPos]
             let argDeclType = argDecl.Type
-            argDeclType.HasTypeDefinition && (getAbbreviatedEntity argDeclType.TypeDefinition).IsDelegate
+            let argIsDelegate =
+                argDeclType.HasTypeDefinition && (getAbbreviatedEntity argDeclType.TypeDefinition).IsDelegate
+
+            if argIsDelegate then
+                let apparentEntity = m.ApparentEnclosingEntity
+                let hasOverload =
```
Comment (auduchinok) [inline ReSharper.FSharp/src/FSharp.Psi.Features/src/Daemon/Analyzers/LambdaAnalyzer.fs:108]: What particular overload this name is about?

**Review by auduchinok — COMMENTED** [review]
Mostly some nit-picking about formatting and naming.

File: ReSharper.FSharp/src/FSharp.Psi.Features/src/Daemon/Analyzers/LambdaAnalyzer.fs (hunk 1 above)
Comment (DedSec256) [reply ReSharper.FSharp/src/FSharp.Psi.Features/src/Daemon/Analyzers/LambdaAnalyzer.fs:108]: any overload corresponding to the criterion

File: ReSharper.FSharp/src/FSharp.Psi.Features/src/Daemon/Analyzers/LambdaAnalyzer.fs [hunk 2]
```diff
@@ -96,11 +96,22 @@ type LambdaAnalyzer() =
         | :? FSharpMemberOrFunctionOrValue as m ->
             m.IsMember &&
             let lambdaPos = if isNotNull appTuple then appTuple.Expressions.IndexOf(argExpr) else 0
-            let args = m.CurriedParameterGroups
-            if args.[0].Count <= lambdaPos then false else
-            let argDecl = args.[0].[lambdaPos]
+            let args = m.CurriedParameterGroups.[0]
```
[the remark this answers is not in the document]
Comment (DedSec256) [reply ReSharper.FSharp/src/FSharp.Psi.Features/src/Daemon/Analyzers/LambdaAnalyzer.fs:99]: I think not, even for properties there will be 1 group of dimension 0

[the remark this answers is not in the document]
Comment (auduchinok) [reply ReSharper.FSharp/src/FSharp.Psi.Features/src/Daemon/Analyzers/LambdaAnalyzer.fs:99]: https://github.com/dotnet/fsharp/blob/3815b8f5d1a0089898428281d2139b0e6f5cd7ed/src/fsharp/symbols/Symbols.fs#L1925

[the remark this answers is not in the document]
Comment (DedSec256) [reply ReSharper.FSharp/src/FSharp.Psi.Features/src/Daemon/Analyzers/LambdaAnalyzer.fs:99]: Yes I was wrong :(

**Review by auduchinok — APPROVED** [review]
Thanks!

**Pull request merged**